services/stats/stats.go

```diff
@@ -212,188 +191,6 @@ func (s *HandleT) NewWriteKeyStat(Name string, StatType string, writeKey string)
-	return DefaultStats.GetRouterStat(Name, StatType, destName, respStatusCode)
-}
-
-/*
-GetProcErrorStat is used to create new destination specific stat.
-Destination name and response status are added as tags in this case.
-If Destination name and response status have been used on this function before, a RudderStats with the same underlying client will be returned.
-*/
-func (s *HandleT) GetProcErrorStat(Name string, StatType string, destName string, statusCode int, stage string) RudderStats {
-	procErrorClientsMapLock.Lock()
-	defer procErrorClientsMapLock.Unlock()
-	key := fmt.Sprintf("%s|%d|%s", destName, statusCode, stage)
-	if _, found := procErrorClientsMap[key]; !found {
-		var err error
-		procErrorClientsMap[key], err = statsd.New(conn, statsd.TagsFormat(statsd.InfluxDB), statsd.Tags("instanceName", instanceID, "destName", destName, "statusCode", strconv.Itoa(statusCode), "stage", stage))
-		if err != nil {
-			pkgLogger.Error(err)
-		}
-	}
-	return &RudderStatsT{
-		Name:        Name,
-		StatType:    StatType,
-		Client:      procErrorClientsMap[key],
-		dontProcess: false,
-	}
-}
-
-// GetProcErrorStat is used to create new destination specific stat.
-// Deprecated: Use DefaultStats for managing stats instead
-func GetProcErrorStat(Name string, StatType string, destName string, statusCode int, stage string) RudderStats {
-	return DefaultStats.GetProcErrorStat(Name, StatType, destName, statusCode, stage)
-}
-
-/*
-NewJobsDBStat is used to create new JobsDB specific stat.
-JobsDB customVal is added as the value of 'customVal' tag in this case.
-If customVal has been used on this function before, a RudderStats with the same underlying client will be returned.
-*/
-func (s *HandleT) NewJobsDBStat(Name string, StatType string, customVal string) RudderStats {
-	jobsdbClientsMapLock.Lock()
-	defer jobsdbClientsMapLock.Unlock()
-	if _, found := jobsdbClientsMap[customVal]; !found {
-		var err error
-		jobsdbClientsMap[customVal], err = statsd.New(conn, statsd.TagsFormat(statsd.InfluxDB), statsd.Tags("instanceName", instanceID, "customVal", customVal))
-		if err != nil {
-			pkgLogger.Error(err)
-		}
-	}
-	return &RudderStatsT{
-		Name:     Name,
-		StatType: StatType,
-		Client:   jobsdbClientsMap[customVal],
-	}
-}
-
-// NewJobsDBStat is used to create new JobsDB specific stat.
-// Deprecated: Use DefaultStats for managing stats instead
-func NewJobsDBStat(Name string, StatType string, customVal string) RudderStats {
-	return DefaultStats.NewJobsDBStat(Name, StatType, customVal)
-}
-
-/*
-NewMigratorStat is used to create new Migrator specific stat.
-Migrator migrationType is added as the value of 'migrationType' tag in this case.
-If migrationType has been used on this function before, a RudderStats with the same underlying client will be returned.
-*/
-func (s *HandleT) NewMigratorStat(Name string, StatType string, migrationType string) RudderStats {
```

Comment:
Keep this function for some more time as it's used in enterprise repo. Just in case if we want to use older enterprise code with newer rudder-server code. Mark it deprecated. We can clean it up later.